### PR TITLE
SPARKC-266 Fixed VarInt Column filter error In Spark SQL

### PR DESCRIPTION
**Pull request by alexliu68**
VarInt column is mapped to Spark Sql Decimal type, the data needs
be converted to decimal to Catalyst. It also needs to be converted
back to BigInteger when push down the filters to Cassandra.

### CODE REVIEW

**Comment by bcantoni** [comment]
Any other doc changes needed? (Probably CHANGES.txt at least.)

**Comment by alexliu68** [comment]
It could be added to data mapping section to describe how VarInt data is stored in Spark SQL

File: spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/sql/CassandraSQLSpec.scala [hunk 1]
```diff
@@ -421,4 +419,33 @@ class CassandraSQLSpec extends SparkCassandraITFlatSpecBase {
...
+        |  PRIMARY KEY ((id, series, rollup_minutes), event)
+        |)
+      """.stripMargin.replaceAll("\n", " "))
+      session.execute(
+        s"""
+        |INSERT INTO sql_test.blob_test(id, series, rollup_minutes, event)
+        |VALUES(1234567891234, 1234567891235, 1234567891236, 'event')
+      """.stripMargin.replaceAll("\n", " "))
+    }
+    val cc = new CassandraSQLContext(sc)
```
Comment (RussellSpitzer) [inline spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/sql/CassandraSQLSpec.scala]: we still have no way of programatically checking if things are pushed down?

File: spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraSourceRelation.scala [hunk 2]
```diff
@@ -125,19 +126,48 @@ private[cassandra] class CassandraSourceRelation(
...
+  /** If column is VarInt column, convert data to BigInteger */
+  private def toCqlValue(columnName: String, value: Any): Any = {
+    if (VarIntColumn(columnName)) {
+      toBigInt(value)
+    } else {
+      value
+    }
+  }
+
+  private def VarIntColumn(columnName: String): Boolean = {
```
Comment (RussellSpitzer) [inline spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraSourceRelation.scala]: possibly name this `isVarInt(columnName:String)`

File: spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraSourceRelation.scala [hunk 3]
```diff
@@ -127,19 +128,48 @@ private[cassandra] class CassandraSourceRelation(
...
+    tableDef.columnByName(columnName).columnType == VarIntType
+  }
+
+  /** Convert decimal to BigInteger */
+  private def toBigInt(value: Any): Any = {
```
Comment (RussellSpitzer) [inline spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraSourceRelation.scala]: why not lock the input of this function to decimal? Also think about naming decimalToBigInt? Then error out if called on non-bigint column

I think the best solution is to make this function and lock input to the type decimal then in toCqlValue do a match on type so the compiler will be happy

Comment (alexliu68) [inline spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraSourceRelation.scala]: non-bigint should return the original value, decimal converts to BigInteger.

There are toCqlValues and toCqlValue both need this function, so if add matching to those function, we duplicate the matching in two places, that's the reason I refactor it into a function.

**Comment by alexliu68** [comment]
@RussellSpitzer I simplify the functions in CassandraSourceRelation, you can check it out

**Comment by RussellSpitzer** [comment]
+1

**Pull request merged**